sympy/parsing/sym_expr.py

```diff
@@ -0,0 +1,250 @@
+from sympy.printing import pycode, ccode, fcode
+from sympy.external import import_module
+lfortran = import_module('lfortran')
+
+if lfortran:
+    from sympy.parsing.fortran.fortran_parser import src_to_sympy
```

Comment:
@asmeurer what should be the best way to handle this? The file will not even import if `lfortran` is not installed. So for now I put this whole file under the conditional `if lfortran:`. Is there a better way?

`sym_expr.py` and `fortran_parser.py` both depend on `lfortran`. So I was thinking of simply importing `lfortran` in those files directly, but making sure that the two modules `sym_expr` and `fortran_parser` are only imported from the test (and nowhere else by default), and the test will be skipped if `lfortran` is not installed. 

Let me know what the best way is.

Reply:
How do the other places in SymPy with `import_module` handle it? I would personally put it in the top-level function if that's possible.

Reply:
They seem to be using the same `if` trick that I did.

Are you against having modules in sympy that will not get imported by default, that would use 3rd party packages, and that the user must import by hand (or when some argument is passed, say `codegen(lang="fortran")`), because there can be a substantial modules that depend on 3rd party packages and it seems ugly and fragile to constantly be checking if a given module is importable.